keel-tagging-plugin/src/main/kotlin/com/netflix/spinnaker/keel/tagging/ResourceTagger.kt

```diff
@@ -83,25 +83,53 @@ class ResourceTagger(
 
   @EventListener(ResourceCreated::class)
   fun onCreateEvent(event: ResourceCreated) {
-    if (event.resourceId.shouldTag()) {
+    if (!event.resourceId.shouldTag()) {
+      return
+    }
+    /**
+     * If a MultiRegion resource provides a set of regional resourceIds, entity tags will be created
+     * against those ids only, not the
+     */
+    if (event.resourceIds.isEmpty()) {
       log.debug("Persisting tag desired for resource {} because it exists now", event.resourceId)
       val spec = event.resourceId.generateKeelTagSpec()
       persistTagState(spec)
     }
+
+    event.resourceIds.forEach { id ->
```

Comment:
I know that if the above condition is true (`resourceIds.isEmpty()`) then this will not run, but it still feels weird to me that it's not in an `else {` block, or there's not a return in the above `if`. Sort of makes me do a double take in a confusing way.